android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsConsentModule.java

```diff
@@ -21,206 +21,134 @@
 import com.facebook.react.bridge.ReactMethod;
-import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.bridge.ReadableMap;
-import com.facebook.react.bridge.WritableArray;
 import com.facebook.react.bridge.WritableMap;
-import com.google.ads.consent.AdProvider;
-import com.google.ads.consent.ConsentForm;
-import com.google.ads.consent.ConsentFormListener;
-import com.google.ads.consent.ConsentInfoUpdateListener;
-import com.google.ads.consent.ConsentInformation;
-import com.google.ads.consent.ConsentStatus;
-import com.google.ads.consent.DebugGeography;
+import com.google.android.ump.ConsentDebugSettings;
+import com.google.android.ump.ConsentInformation;
+import com.google.android.ump.ConsentRequestParameters;
+import com.google.android.ump.UserMessagingPlatform;
 import io.invertase.googlemobileads.common.ReactNativeModule;
-import java.net.MalformedURLException;
-import java.net.URL;
 import java.util.List;
 
 public class ReactNativeGoogleMobileAdsConsentModule extends ReactNativeModule {
+
   private static final String TAG = "RNGoogleMobileAdsConsentModule";
   private ConsentInformation consentInformation;
-  private ConsentForm consentForm;
 
   public ReactNativeGoogleMobileAdsConsentModule(ReactApplicationContext reactContext) {
     super(reactContext, TAG);
-    consentInformation = ConsentInformation.getInstance(reactContext);
+    consentInformation = UserMessagingPlatform.getConsentInformation(reactContext);
   }
 
-  private int getConsentStatusInt(ConsentStatus consentStatus) {
+  private String getConsentStatusString(int consentStatus) {
     switch (consentStatus) {
-      case NON_PERSONALIZED:
-        return 1;
-      case PERSONALIZED:
-        return 2;
-      case UNKNOWN:
+      case ConsentInformation.ConsentStatus.REQUIRED:
+        return "REQUIRED";
+      case ConsentInformation.ConsentStatus.NOT_REQUIRED:
+        return "NOT_REQUIRED";
+      case ConsentInformation.ConsentStatus.OBTAINED:
+        return "OBTAINED";
+      case ConsentInformation.ConsentStatus.UNKNOWN:
       default:
-        return 0;
+        return "UNKNOWN";
     }
   }
 
   @ReactMethod
-  public void requestInfoUpdate(ReadableArray publisherIds, Promise promise) {
-    @SuppressWarnings("SuspiciousToArrayCall")
-    String[] publisherIdsArray = publisherIds.toArrayList().toArray(new String[0]);
-
-    consentInformation.requestConsentInfoUpdate(
-        publisherIdsArray,
-        new ConsentInfoUpdateListener() {
-          @Override
-          public void onConsentInfoUpdated(ConsentStatus consentStatus) {
+  public void requestInfoUpdate(final ReadableMap options, final Promise promise) {
+    try {
+      ConsentRequestParameters.Builder paramsBuilder = new ConsentRequestParameters.Builder();
+      ConsentDebugSettings.Builder debugSettingsBuilder =
+          new ConsentDebugSettings.Builder(getApplicationContext());
+
+      if (options.hasKey("testDeviceIdentifiers")) {
```

Comment:
This is a nit-pick, but we're dereferencing options here without noting it is `@Nonnull` as a parameter annotation or testing it for null. Further on I can see at the typescript level you're verifying it is an object so this is probably safe and just a code style thing. I'd go with a `@Nonnull` annotation from androidx.annotations normally then, or leave it if that doesn't work